### DISCUSSION TITLE
v4.0.1

### DISCUSSION THREAD

**Discussion by InfinityLoop1308** [InfinityLoop1308/PipePipe#555] [Announcements]
### New Features

- Redesign the settings to make them clearer
- Support more languages: fr, de, it, es

### Bug Fixes

- Sign in and out does not take effect immediately
- Just ended streams can not be viewed (YouTube)
- Sign in does not work due to new blocking (YouTube)
- Live chat not available (YouTube)
- Trending not available for ip outside japan (NicoNico)

### Others

- Improve stability 

<hr/>
If you're enjoying PipePipe, please help share it with your friends. Every bit of support means a lot to us!
<hr/>

### Which APK should you download?

- arm64-v8a should work on most devices.
- armeabi-v7a if your device is a 10+ year old phone.
- x86_64 for 64-bit Android simulators (on a PC).
- x86 for 32-bit simulators.

If these rules not working, download the universal APK. It should work on any device.
All APKs provide the same contents and functions, but the universal one may be slower than the split ones.

<hr /><em>This discussion was created from the release <a href='https://github.com/InfinityLoop1308/PipePipe/releases/tag/'>v4.0.1</a>.</em>

**Comment by yurenchen000** [comment]
thanks for your work! ✨

the arm64 apk size increase 13MB than previous  
i wonder what happened

**Reply by InfinityLoop1308** [reply]
yt-dlp is bundled to avoid being blocked by YouTube

**Comment by flanter21** [comment]
Hi, I was wondering where the default video and audio formats can be changed in this version. It's no longer under Video and Audio like previous versions and I can't find it. Thank you.

**Reply by InfinityLoop1308** [reply]
Now PipePipe will always try to select the format with the highest bitrate at the current resolution. The order is usually av1-encoded mp4 > webm > vp9-encoded mp4. For compatibility reasons, av1 and webm are disabled by default. If your device supports them, you can enable them in Advanced settings to save data usage (4.1.0).

In 4.0.1, we used webm by default in almost all places, which caused some compatibility issues - 4.1.0 will resolve this issue. You can try this [latest build version](https://github.com/InfinityLoop1308/PipePipe/actions/runs/12337341516).

**Comment by Sifattoll** [comment]
আসসালামু আলাইকুম
